### DISCUSSION TITLE
Proposal: BPM counter

### DISCUSSION THREAD

**Discussion by meT1X123** [ppy/osu#12994] [Ideas] [answered]
1) Create a BPM counter that would show the BPM of the running map.
For example, during the game you thought about something, and now there will be a stream parts, and you do not remember the BPM anymore, and then you have a BPM counter (displayed, for example, under "K1, K2, M1, M2").
Ways of switching on:
a) Through settings > BPM counter > "Show BPM all the time while playing the map" or "Show BPM during a break on the map".
or
b) Through any key configured in the controls, but you can switch mode only when you play > "Show BPM all the time while playing the map" or "Show BPM during a break on the map".

2) Create a BPM counter that would show the BPM of tapping "K1", "K2" or "M1", "M2" (every tap makes an refresh the BPM counter) while you are watching your or another person's replay. For example, to see where you or another person understreaming or overstreaming, and understand the problem.
Ways of switching on: same.
P.S. Why for the second proposal to add the ability to turn the BPM counter on and off? Fps on weak computers may decrease (less 60fps) because of the refresh of the BPM counter for every tap while watching a replay. But I'm not sure that refresh of the BPM counter for every tap make fps less.

**Comment by smoogipoo** [comment]
I highly disagree with creating a BPM counter for the beatmap, because part of the rhythm of a rhythm game is feeling the BPM - just seeing a visual counter does nothing imo.

Agree with adding a keys-per-second counter or similar, though.

**Comment by peppy** [comment]
I'm not against having some kind of display which shows changes in SV, BPM etc. if it was designed to be visually appealing. Having it fade in just before the change, then counting up to the new value and transitioning out could be a pretty nice effect.

Having it permanently displayed seems pointless and will just add clutter.

**Comment by mk56-spn** [comment, marked as answer]
will be closed by: https://github.com/ppy/osu/pull/21519
